list_api/mock_list_api_failure.py

```diff
@@ -0,0 +1,139 @@
+import random
+import time
+
+import out
+
+from .models import Course, Problem, Submit, ListSession
+
+chance_to_success = 0.0
+
+
+def login(email: str, password: str) -> ListSession:
+    time.sleep(1)
+    if random.random() > chance_to_success:
+        out.error("Login failed, bad server status code: 500", "Login failed")
```

Comment:
the context paramter is for some "contextual" information, no for the same thing as the message. You don't have to use it all the time. It is for something like the causing exception etc.